R/include-diabetes-diagnoses.R

```diff
@@ -68,22 +71,19 @@ include_diabetes_diagnoses <- function(lpr2, lpr3) {
       .keep = "all",
       .by = "pnr"
     ) |>
-    # Coalesce NA values to 0
+    # Convert NA values to 0
     dplyr::mutate(
-      n_t1d_endocrinology = dplyr::coalesce(.data$n_t1d_endocrinology, 0),
-      n_t2d_endocrinology = dplyr::coalesce(.data$n_t2d_endocrinology, 0),
-      n_t1d_medical = dplyr::coalesce(.data$n_t1d_medical, 0),
-      n_t2d_medical = dplyr::coalesce(.data$n_t2d_medical, 0),
-      .by = "pnr"
+      dplyr::across(
+        tidyselect::matches("^n_t[12]d_"),
+        \(x) dplyr::coalesce(x, 0)
+      )
     ) |>
     # Keep earliest two dates per individual.
     dplyr::filter(dplyr::row_number(.data$date) %in% 1:2, .by = "pnr") |>
+    dplyr::mutate(has_majority_t1d_diagnosis = !!logic) |>
```

Comment:
```suggestion
    dplyr::mutate(has_majority_t1d_dx = !!logic) |>
```
? :) so it's aligned with`is_primary_dx`